lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -33,12 +33,12 @@ def initialize(options={})
       end
 
       def purchase(money, payment, options={})
-        options[:idempotency_key] = nil
         if options[:execute_threed] || options[:threed_dynamic]
           authorize(money, payment, options)
         else
           MultiResponse.run do |r|
             r.process { authorize(money, payment, options) }
+            options[:idempotency_key] = "#{options[:idempotency_key]}-cap" if options[:idempotency_key]
```

Comment:
@jknipp @bayprogrammer The purchase action mutates `options[:idempotency_key]` after `authorize` so that the `capture` request has a unique key. This is why I re-set the key before the second purchase request in the remote test.

I chose to modify it in the exact same way each time, so that repeat purchase requests will lead to a non-unique key for the `capture`. Then the entire purchase action can be repeated without risk of duplication.